### DISCUSSION TITLE
How to run without docker desktop for Linux

### DISCUSSION THREAD

**Discussion by papanada** [wardenenv/warden#727] [Help & Support]
### Version of Warden

0.14.1

### Operating System and Installation Method

Debian12 via brew

### Describe the Bug

Docker Desktop for linux has severe permissions problems that makes warden unusable.
There is a way to run warden without having docker desktop installed?

**Comment by bap14** [comment]
Warden requires a docker services. It doesn't _have_ to be Docker Desktop, you can replace it with something like OrbStack for MacOS, Lima or Colima, containerd with nerdctl. Now you may have to add some aliases if the solution doesn't automatically have an alias for the "docker" command.

That being said, I'd like to hear what permissions problems you're having with Docker Desktop and Warden.

**Comment by papanada** [comment]
@bap14 
Docker desktop changes project folder's ownership.
It sets as 100999:100999
I'have tryed to add my user to that group ID but I can't still have write permission on the folder.

Can't I run warden with docker engine only?
(I'm on Debian12 with Cinnamon)

**Comment by bap14** [comment]
@ortoDev Warden requires a Docker engine that is compatible with the standard Docker commands (`docker ps`, `docker compose`, `docker network`, etc.). Docker Desktop is just the simplest solution for non-linux environments.

**Comment by papanada** [comment]
Thanks @bap14 
Now I have a running warden svc on my machine only with docker engine.
But when I try to reach http://mailhog.warden.test/ it gives me "Server not found".
I've already mapped *test to 127.0.0.1 on my hosts file.

I need to add manually every host in my hosts file to see them in browser

**Comment by bap14** [comment]
@ortoDev So if you go to https://traefik.warden.test do you see the Traefik dashboard? If not, then I'd ask you to run `dig traefik.warden.test` to ensure that returns as 127.0.0.1. If not, then you have an issue with your resolver setup that you'd need to investigate further.

**Comment by riconeitzel** [comment]
I'm working with colima on a mac and I'm unable to install docker-compose-plugin … I tried setting the DOCKER_COMPOSE_COMMAND to "hack" in the "old" version 2 of `docker-compose`. Unfortunately that didn't do the trick.

There is no way (at least I didn't find any in the last 45 minutes) to install the docker-compose-plugin via homebrew. Now I'm a bit stuck 😄 Any suggestions from anyone who had the same issue? 🤞

**Reply by navarr** [reply]
The current version of Warden uses `docker compose` - as compose is now built in to Docker.  Are you getting errors from Warden with the defaults set?

**Reply by riconeitzel** [reply]
yes I did … I just installed `warden` today (0.14.2) so I assume that's the latest version. docker-compose is also the latest version. I just stumbled upon this post here: https://smallsharpsoftwaretools.com/tutorials/use-colima-to-run-docker-containers-on-macos/

and the actual solution simply was to symlink docker-compose to the cli directory:

`ln -sfn $(brew --prefix)/opt/docker-compose/bin/docker-compose ~/.docker/cli-plugins/docker-compose`

Also hacking `warden` did work but was not an option 😂
